src/duration.rs

```diff
@@ -742,38 +742,76 @@ impl Duration {
 // region: trait impls
 /// The format returned by this implementation is not stable and must not be relied upon.
 ///
+/// By default this produces an exact, verbose printout of the duration.
+/// For a concise, rounded printout instead, you can use the `.N` format specifier:
+///
+/// ```
+/// use time::Duration;
+///
+/// let duration = Duration::new(123456, 789011223);
+/// println!("{:.3}", duration);
+/// ```
+///
 /// For the purposes of this implementation, a day is exactly 24 hours and a minute is exactly 60
 /// seconds.
 impl fmt::Display for Duration {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        /// Format a single item.
-        macro_rules! item {
-            ($name:literal, $value:expr) => {
-                match $value {
-                    0 => Ok(()),
-                    value => value.fmt(f).and(f.write_str($name)),
-                }
-            };
-        }
-
         if self.is_zero() {
             return f.write_str("0s");
```

Comment:
If the user is requesting a certain number of decimal places, we should always give them that many digits. I think this could be done by formatting `0f32` and appending the "s"`. It should still be "0s" by default, though.